test/adapter/json_api/pagination_links_test.rb

```diff
@@ -21,30 +21,70 @@ def setup
           ]
         end
 
-        def mock_request(query_parameters = {}, original_url = URI)
+        def test_pagination_links_using_kaminari
+          assert_links_serialization(expected_response_with_pagination_links, kaminari_collection, options)
+        end
+
+        def test_pagination_links_using_will_paginate
+          assert_links_serialization(expected_response_with_pagination_links, will_paginate_collection, options)
+        end
+
+        def test_pagination_links_with_additional_params
+          assert_links_serialization(expected_response_with_pagination_links_and_additional_params, will_paginate_collection, options(test: 'test'))
+        end
+
+        def test_pagination_config_set_to_false
+          previous_config = ActiveModelSerializers.config.collection_serializer
+          ActiveModelSerializers.config.collection_serializer = ActiveModel::Serializer::NonPaginatedCollectionSerializer
+          assert_links_serialization(paginated_data, kaminari_collection, options)
+        ensure
+          ActiveModelSerializers.config.collection_serializer = previous_config
+        end
+
+        def test_not_showing_pagination_links
+          assert_links_serialization(non_paginated_data, @array)
+        end
+
+        def test_non_paginated_serializer
+          assert_links_serialization(paginated_data, kaminari_collection, options.merge(serializer: ActiveModel::Serializer::NonPaginatedCollectionSerializer))
+        end
+
+        def test_last_page_pagination_links_using_kaminari
+          assert_links_serialization(expected_response_with_last_page_pagination_links, kaminari_collection(3), options)
+        end
+
+        def test_last_page_pagination_links_using_will_paginate
+          assert_links_serialization(expected_response_with_last_page_pagination_links, will_paginate_collection(3), options)
+        end
+
+        private
+
+        def options(query_parameters = {}, original_url = URI)
```

Comment:
I liked how the `mock_request` helper declared what was going on. This kind of hides important info of what the test is doing